api/metrics.go

```diff
@@ -62,32 +64,32 @@ func metricsMiddleware(next http.Handler) http.Handler {
 		var (
 			enabled      = false
 			name         = ""
-			subscription = ""
+			subscription = false
 		)
 
 		// all named route will be recorded
 		if rt != nil && rt.GetName() != "" {
 			enabled = true
 			name = rt.GetName()
 			if strings.HasPrefix(name, "subscriptions") {
-				// example path: /subscriptions/txpool -> subject = txpool
-				paths := strings.Split(r.URL.Path, "/")
-				if len(paths) > 2 {
-					subscription = paths[2]
-				}
+				subscription = true
+				name = "WS " + r.URL.Path
 			}
 		}
 
 		now := time.Now()
 		mrw := newMetricsResponseWriter(w)
-		if subscription != "" {
-			metricActiveWebsocketCount().AddWithLabel(1, map[string]string{"subject": subscription})
+		if subscription {
+			metricActiveWebsocketGauge().AddWithLabel(1, map[string]string{"name": name})
+			metricWebsocketCounter().AddWithLabel(1, map[string]string{"name": name})
 		}
 
 		next.ServeHTTP(mrw, r)
 
-		if subscription != "" {
-			metricActiveWebsocketCount().AddWithLabel(-1, map[string]string{"subject": subscription})
+		if subscription {
+			metricActiveWebsocketGauge().AddWithLabel(-1, map[string]string{"name": name})
+			// record websocket duration in seconds, not MS
+			metricWebsocketDuration().ObserveWithLabels(time.Since(now).Milliseconds() / 1000, map[string]string{"name": name, "code": strconv.Itoa(mrw.statusCode)})
```

Comment:
This allows us to capture error codes + websocket durations (in seconds)

<img width="942" alt="image" src="https://github.com/user-attachments/assets/3086c325-2946-4550-bf15-bccb33c429e5">